scipy/io/mmio.py

```diff
@@ -281,20 +286,44 @@ def _get_symmetry(a):
         m,n = a.shape
         if m != n:
             return MMFile.SYMMETRY_GENERAL
-        issymm = 1
-        isskew = 1
+        issymm = True
+        isskew = True
         isherm = a.dtype.char in 'FD'
-        for j in range(n):
-            for i in range(j+1,n):
-                aij,aji = a[i][j],a[j][i]
-                if issymm and aij != aji:
-                    issymm = 0
-                if isskew and aij != -aji:
-                    isskew = 0
-                if isherm and aij != conj(aji):
-                    isherm = 0
-                if not (issymm or isskew or isherm):
-                    break
+        
+        if isinstance(a,spmatrix):
+            # check if number of nonzero entrys of lower and upper triangle matrix are equal
+            (row, col) = a.nonzero()
+            if (row < col).sum() != (row > col).sum():
+                return MMFile.SYMMETRY_GENERAL
+            
```

Comment:
Yes, my code does assume that `a.nonzero()` returns rows, columns, with the row indices indices sorted by value, and the column indices sorted by value within row.   All the examples I have tried have this property, but now I look at the docstring, I don't see any comment on that.  Have you come across a counter-example?    Does anyone know whether this is a guarantee?